engine/schema/src/com/cloud/storage/dao/VMTemplateZoneDaoImpl.java

```diff
@@ -84,14 +84,11 @@ public VMTemplateZoneVO findByZoneTemplate(long zoneId, long templateId) {
     }
 
     @Override
+    @DB
     public void deletePrimaryRecordsForTemplate(long templateId) {
         SearchCriteria<VMTemplateZoneVO> sc = TemplateSearch.create();
         sc.setParameters("template_id", templateId);
-        TransactionLegacy txn = TransactionLegacy.currentTxn();
```

Comment:
I looked a bit more into that, and `@DB` use the same class as `TransactionLegacy`, which is different than `Transaction`. The change f62e28c1ec125263613d32856b11ef50b815e573 added a new transaction API.

Having the annotation or the line of code to create the transaction is similar. IMO the annotation is cleaner.